### PR TITLE
fix(core): add route name as tx type for consolidate/fanout

### PR DESCRIPTION
**Pull request by tylerlevine**
Explicitly tag these special types of self-sends, and only call the
specialized build routes for these types when that tag is set.

This prevents inadvertent consolidations due to approving a pending
approval with no recipients (self-sends, since only external recipients
appear in the recipients list).

Note: fanout pending approvals won't work correctly, since their
specialized build route is not implemented in the pending approval
rebuild process.

Ticket: BG-29732

### CODE REVIEW

**Comment by tylerlevine** [comment]
This is somewhat difficult to test, since it involves a lot of moving pieces.

Here's the `transactionRequest` part of the pending approval for a consolidation transaction I did using this branch:
```
transactionRequest:
         { coinSpecific:
            { tltc:
               { txHex:
                  '01000000039366b437c46263f845e9659bc68cec303011d44300fb031cfa573d41916bd67e00000000b60047304402204693289d50cad7d888545f0c5bcc01b6392214a647ff6b4a53ad889e9c937aa40220089b3e1f6ab06d243ebeccc97d8ff6a62e69cf825a3e70483ebd77c2f3e16b1b0100004c69522103a53df33c9c2221509237e5f4a9a28f723751a1f51f2d816bbf5be288ad33b8ba21038fdbeb54cc1fa3ec40b2bd7d19a46987fd3206689e09a5831be10e53b20804f321020932b9c3ae445644387afe27314d1106ba9490619fb5df95c0da80fbb549f5ae53aeffffffff7337f9454abb564699be228b6f9b800460707df9410caf1d322cead41f77c0ac00000000b700483045022100bd4b9a7dc7268d98052df7368d4acd3e9ffb2e312fed9d62d965b4ed286a31ff022064d92b27f21d1db52d5cff63b789729e12257e74285bdcf67e1a1c0d28cac6fa0100004c69522103a53df33c9c2221509237e5f4a9a28f723751a1f51f2d816bbf5be288ad33b8ba21038fdbeb54cc1fa3ec40b2bd7d19a46987fd3206689e09a5831be10e53b20804f321020932b9c3ae445644387afe27314d1106ba9490619fb5df95c0da80fbb549f5ae53aeffffffff8ff9794b19ed444dbe90a78882e58baf12bf74e28d05af36e9defd79823965d000000000b600473044022027f8be2f834152120153eba0afb847420b4d212dbb9dbdd668064be6ae28a81a02206b342ffde49471bc12583a869be3fc712dd6c651d1ea75e520e3076b378897db0100004c69522102cac40c28d6a36b5e7bb0224be429b5d5ed8779c19e611c5f7f3cf2da8af5f6fb2103ae0066f7a4f69d705eee3e2d2fc0480f6891f18e0d33e296bb5f2706e8c550fd2103be658bb8c6bf3a31ee149ccb252df3e8c9e4bc3da3eb569ddcef28af2378725953aeffffffff01e5cabc270000000017a9140daaf2ffe9beab3b42346bdd23c13e94585294678700000000' } },
           verificationItems: [],
           videoApprovers: [],
           requestedAmount: '0',
           fee: 933,
           sourceWallet: '5b3bd83d032bc0ad03bee3bd2a807c57',
           recipients: [],
           buildParams: { type: 'consolidate' },
           policyUniqueId: '5b3bd83d032bc0ad03bee3be' } },
```

`type` does appear and is set correctly, so looks like pending approvals created from consolidation transactions are working as expected.

I wrote another script to approve the consolidation pending approval, and that works as well. Here's the result from the approval:
```
{ id: '6052a09e9276900032943383a879210c',
  coin: 'tltc',
  wallet: '5b3bd83d032bc0ad03bee3bd2a807c57',
  creator: '5a56b9c58f90ac3f074089909026a1da',
  createDate: '2021-03-18T00:36:46.930Z',
  info:
   { type: 'transactionRequest',
     transactionRequest:
      { coinSpecific:
         { tltc:
            { txHex:
               '01000000039366b437c46263f845e9659bc68cec303011d44300fb031cfa573d41916bd67e00000000b60047304402204693289d50cad7d888545f0c5bcc01b6392214a647ff6b4a53ad889e9c937aa40220089b3e1f6ab06d243ebeccc97d8ff6a62e69cf825a3e70483ebd77c2f3e16b1b0100004c69522103a53df33c9c2221509237e5f4a9a28f723751a1f51f2d816bbf5be288ad33b8ba21038fdbeb54cc1fa3ec40b2bd7d19a46987fd3206689e09a5831be10e53b20804f321020932b9c3ae445644387afe27314d1106ba9490619fb5df95c0da80fbb549f5ae53aeffffffff7337f9454abb564699be228b6f9b800460707df9410caf1d322cead41f77c0ac00000000b700483045022100bd4b9a7dc7268d98052df7368d4acd3e9ffb2e312fed9d62d965b4ed286a31ff022064d92b27f21d1db52d5cff63b789729e12257e74285bdcf67e1a1c0d28cac6fa0100004c69522103a53df33c9c2221509237e5f4a9a28f723751a1f51f2d816bbf5be288ad33b8ba21038fdbeb54cc1fa3ec40b2bd7d19a46987fd3206689e09a5831be10e53b20804f321020932b9c3ae445644387afe27314d1106ba9490619fb5df95c0da80fbb549f5ae53aeffffffff8ff9794b19ed444dbe90a78882e58baf12bf74e28d05af36e9defd79823965d000000000b600473044022027f8be2f834152120153eba0afb847420b4d212dbb9dbdd668064be6ae28a81a02206b342ffde49471bc12583a869be3fc712dd6c651d1ea75e520e3076b378897db0100004c69522102cac40c28d6a36b5e7bb0224be429b5d5ed8779c19e611c5f7f3cf2da8af5f6fb2103ae0066f7a4f69d705eee3e2d2fc0480f6891f18e0d33e296bb5f2706e8c550fd2103be658bb8c6bf3a31ee149ccb252df3e8c9e4bc3da3eb569ddcef28af2378725953aeffffffff01e5cabc270000000017a9140daaf2ffe9beab3b42346bdd23c13e94585294678700000000' } },
        verificationItems: [],
        videoApprovers: [],
        requestedAmount: '0',
        fee: 933,
        sourceWallet: '5b3bd83d032bc0ad03bee3bd2a807c57',
        recipients: [],
        buildParams: { type: 'consolidate' },
        policyUniqueId: '5b3bd83d032bc0ad03bee3be',
        validTransaction:
         '01000000039366b437c46263f845e9659bc68cec303011d44300fb031cfa573d41916bd67e00000000fdfd0000483045022100cb283659bd88e39ce2cad1d5ffc43437ebfaa30f76e3a50890eeba6f0b6e162802206592a91f800b8d36ed3806474463f030d487a443bf64b37e3fa417a86ac2906d0147304402205c96e2cc270c32c9e22efb051f78d49f6ce97afd67b9e997e94ef7a97680dc24022038ff2408cf6938366852891f1dbefe8af6a72603e2077819d8eb517aa275f953014c69522103a53df33c9c2221509237e5f4a9a28f723751a1f51f2d816bbf5be288ad33b8ba21038fdbeb54cc1fa3ec40b2bd7d19a46987fd3206689e09a5831be10e53b20804f321020932b9c3ae445644387afe27314d1106ba9490619fb5df95c0da80fbb549f5ae53aeffffffff7337f9454abb564699be228b6f9b800460707df9410caf1d322cead41f77c0ac00000000fc0047304402200997c19e9fa4e315076154732d3cacf3b67331488e113a547b03bb43258892df02200e3ec4b9f95cf867471d9a3406bece547fc34f43f09e988c681a5ec06b611f31014730440220645baf1e0a81d3ecd0c7d0a358b89559450e69191adb1617addd016d5ce4065202202a3558c74e362542091c99b7178f6db948857434ea4ba09d91fd428d0313635b014c69522103a53df33c9c2221509237e5f4a9a28f723751a1f51f2d816bbf5be288ad33b8ba21038fdbeb54cc1fa3ec40b2bd7d19a46987fd3206689e09a5831be10e53b20804f321020932b9c3ae445644387afe27314d1106ba9490619fb5df95c0da80fbb549f5ae53aeffffffff8ff9794b19ed444dbe90a78882e58baf12bf74e28d05af36e9defd79823965d000000000fdfd00004830450221008524b3e44cb8e0e3717d0b0bff20a13db1e595b64cbbe08a324fac4af32f2db40220563c2baba7cbc9b65676e80267c221c282c5da2f04d218639a34185eab118a8b01473044022027dda36cb009c49de74336a0987150aa179f01500725a4f6ef37ce4b3c5a404a02203189b59d7d63a861259ca6bc50f7ff24b164181c4acf19b0bffb7535218fa7fe014c69522102cac40c28d6a36b5e7bb0224be429b5d5ed8779c19e611c5f7f3cf2da8af5f6fb2103ae0066f7a4f69d705eee3e2d2fc0480f6891f18e0d33e296bb5f2706e8c550fd2103be658bb8c6bf3a31ee149ccb252df3e8c9e4bc3da3eb569ddcef28af2378725953aeffffffff01e5cabc270000000017a91476571db89c429e56eda24a938797ca2de6a90d948700000000',
        validTransactionHash:
         '023e092efb4553cc1d160291c8c2c91c3a69b6ece607e2db780a478d593ff4f4' } },
  state: 'approved',
  scope: 'wallet',
  signingPolicy: { gatheredSignatures: [] },
  userIds:
   [ '5a56b9c58f90ac3f074089909026a1da',
     '5a55050ba0fde4e2076ffb857c62f47d',
     '5db775e79d87fa3306a94526f6992684' ],
  approvalsRequired: 1,
  singleRunResults: [],
  resolvers:
   [ { user: '5db775e79d87fa3306a94526f6992684',
       date: '2021-03-18T00:53:07.284Z',
       resolutionType: 'pending',
       signatures: [] } ] }
```

**Review by twtaylorbitgo — APPROVED** [review]
image

**Pull request merged**
